activerecord/lib/active_record/connection_adapters/deduplicable.rb

```diff
@@ -5,6 +5,13 @@ module ConnectionAdapters # :nodoc:
     module Deduplicable
       extend ActiveSupport::Concern
 
+      cattr_accessor :classes
+      self.classes = []
```

Comment:
Any non-test reason to keep the list of objects in memory? That seems unnecessary overhead just to be able to test something in the framework.

Reply:
It is just for tests; there doesn't appear to be a way to use e.g. `ObjectSpace` to find all classes with a module included; any ideas on how to achieve the same result without modifying the class (here or in test)?

Reply:
How about listing all the classes we care about in the test instead of dynamicly trying to find? I know it isn't future proof, but at the moment it is only 4 classes that we care about

Reply:
Made this change. The set of classes can vary by which adapter is being tested, and we have to think about subclasses for `Column`, but e.g. when PostgreSQL is being tested here's the list that results:

```
[ActiveRecord::ConnectionAdapters::SqlTypeMetadata, ActiveRecord::ConnectionAdapters::Column, ActiveRecord::ConnectionAdapters::PostgreSQL::Column, ActiveRecord::ConnectionAdapters::NullColumn, ActiveRecord::ConnectionAdapters::PostgreSQL::TypeMetadata] 
```